src/main/java/org/simulator/fba/COBRAsolver.java

```diff
@@ -246,10 +247,13 @@ public COBRAsolver(SBMLDocument doc) throws SBMLException, ModelOverdeterminedEx
 			} else {
 				for (Pair<String, Double> pair : species2Reaction.get(species.getId())) {
 					weights[reaction2Index.get(pair.getKey())] = pair.getValue();
+					numberOfReactions++;
 				}
 			}
 			
-			problem.addConstraint(new LinearEqualsConstraint(weights, 0.0, "cnstrt_" + species.getId()));
```

Comment:
Could you please elaborate a bit more what the problem is when a species participates in only one reaction?

Reply:
@draeger, when the species participates in only one reaction (let it be R1), then the constraint R1 = 0 will be added to the linear problem which does not make sense and also, leads to the wrong solution.

I found this error while running the FBA for the example given in the FBC specification document.